eventmesh-connector-plugin/eventmesh-connector-standalone/src/main/java/org/apache/eventmesh/connector/standalone/broker/MessageQueue.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.eventmesh.connector.standalone.broker;
+
+import com.google.common.base.Preconditions;
+import org.apache.eventmesh.connector.standalone.broker.model.MessageEntity;
+
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * This is a block queue, can get entity by offset.
+ * The queue is a FIFO data structure.
+ */
+public class MessageQueue {
+
+    public MessageEntity[] items;
+
+    private int takeIndex;
+
+    private int putIndex;
+
+    private int count;
+
+    private ReentrantLock lock;
```

Comment:
maybe the lock is final is better in this queue.

Reply:
Thanks, agree with you.